### PR TITLE
fix(cve): Fix CVE scanning in images containing Jar files

### LINKED ISSUE #1428: [Bug] CVE scanning failures: Investigate failure to scan images containing jar files
```
2023-05-05T19:25:41.441Z        INFO    JAR files found
2023-05-05T19:25:41.441Z        INFO    Java DB Repository: :1
2023-05-05T19:25:41.441Z        INFO    Downloading the Java DB...
2023-05-05T19:25:41.442Z        ERROR   Unable to initialize the Java DB: Java DB update failed: Java DB update error: DB download error: repository name error (:1): could not parse reference: :1
{"level":"error","error":"scan error: scan failed: failed analysis: analyze error: failed to analyze layer (sha256:7399365e40c41fb2a2fd7321a945b6076c8a2f08e96c114993ea7b25d7660e92): post analysis error: post analysis error: Java DB update failed: Java DB update error: DB download error: repository name error (:1): could not parse reference: :1","image":"public/c3/ubuntu/openj9-amd64:11","goroutine":3819,"caller":"zotregistry.io/zot/pkg/extensions/search/cve/trivy/scanner.go:272","time":"2023-05-05T19:25:41.442428743Z","message":"unable to scan image"}
```

### PR DESCRIPTION
**Pull request by andaaron**
Closes #1428

Looks like Trivy uses a separate DB for jar scanning. This DB is very large and takes additional time to download It has 430 MB at download time, and will grow afterwards

sqlite is a dependency for trivy to use this DB

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -117,8 +117,14 @@ privileged-test: check-skopeo $(TESTDATA)
...
 $(TESTDATA): check-skopeo
-	$(shell mkdir -p ${TESTDATA}; cd ${TESTDATA}; mkdir -p noidentity; ../scripts/gen_certs.sh; cd ${TESTDATA}/noidentity; ../../scripts/gen_nameless_certs.sh; cd ${TOP_LEVEL}; skopeo --insecure-policy copy -q docker://public.ecr.aws/t0x7q1g8/centos:7 oci:${TESTDATA}/zot-test:0.0.1;skopeo --insecure-policy copy -q docker://public.ecr.aws/t0x7q1g8/centos:8 oci:${TESTDATA}/zot-cve-test:0.0.1)
-	$(shell chmod -R a=rwx ${TESTDATA})
+	mkdir -p ${TESTDATA}; \
+	cd ${TESTDATA}; ../scripts/gen_certs.sh; \
+	mkdir -p noidentity; cd ${TESTDATA}/noidentity; ../../scripts/gen_nameless_certs.sh; \
+	cd ${TOP_LEVEL}; \
+	skopeo --insecure-policy copy -q docker://public.ecr.aws/t0x7q1g8/centos:7 oci:${TESTDATA}/zot-test:0.0.1; \
+	skopeo --insecure-policy copy -q docker://public.ecr.aws/t0x7q1g8/centos:8 oci:${TESTDATA}/zot-cve-test:0.0.1; \
+	skopeo --insecure-policy copy -q -f oci docker://jetty:9.4.51-jre8-alpine-eclipse-temurin oci:${TESTDATA}/zot-cve-java-test:0.0.1; \
```
Comment (andaaron) [inline Makefile:126]: @rchincha we need an image containing jar files uploaded to `docker://public.ecr.aws/t0x7q1g8`

File: pkg/extensions/extension_search.go [hunk 2]
```diff
@@ -77,6 +66,7 @@ func EnableSearchExtension(config *config.Config, storeController storage.StoreC
 func downloadTrivyDB(interval time.Duration, sch *scheduler.Scheduler, cveInfo CveInfo, log log.Logger) {
 	generator := NewTrivyTaskGenerator(interval, cveInfo, log)
 
+	log.Info().Msg("Submitting CVE DB update scheduler")
```
Comment (andaaron) [inline pkg/extensions/extension_search.go:69]: Do not use scheduler for first download.

Comment (andaaron) [reply pkg/extensions/extension_search.go:69]: We'll keep using it

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**